paper/main.tex

```diff
@@ -924,107 +938,103 @@ \section{Evaluation of performance}\label{section:evaluation_of_performance}
+literature. The \numberofstrategies strategies are drawn from the APL and include
+strategies from the IPD literature. The computer tournaments include tournaments of
+four different types.
+
+So what is the best way of playing the IPD? And is
+there a single dominant strategy for the IPD? 
+
+There was not a single strategy
+within the collection of the \numberofstrategies strategies that managed to
+perform well in all the tournaments variations it competed in. Even if on
+average a strategy ranked highly in a specific environment this did not
+guarantee its success over the different tournament types. Nevertheless, in
+Sections~\ref{section:top_performances}
+and~\ref{section:evaluation_of_performance} we examined the best performing
+strategies across various tournament types and analysed their salient features.
+this demonstrated that there are properties associated with the success of
+strategies which in fact contradict the originally suggested properties of R.
+Axelrod~\cite{Axelrod1981}.
+
+We showed that complex or \textbf{clever} strategies can be effective, whether
+trained against a corpus of possible opponents or purposely designed to mitigate
+the impact of noise such as the DBS strategy. Moreover, we found some
+strategies designed or trained for noisy environments were also highly ranked in
+noise-free tournaments which reinforces the idea that strategies'
+complexity/cleverness is not necessarily a liability, rather it can confer
+adaptability to a more diverse set of environments.
+We also showed that while the type of exploitation attempted by ZDs is
+not typically effective in standard tournaments, \textbf{envious} strategies
+capable of both exploiting and not their opponents can be highly successful.
+Based on the results of~\cite{Harper2017} this could be because they are
+selectively exploiting weaker opponents while mutually cooperating with stronger
+opponents. Highly noisy or tournaments with short matches also favoured envious
+strategies. These environments mitigated the value of being nice. Uncertainty
+enables exploitation, reducing the ability of maintaining or enforcing mutual
+cooperation, while triggering grudging strategies to switch from typically
+cooperating to typically defecting.
+
+The features analysis of the best performing strategies demonstrated that a
+strategy should reciprocate, as suggested by R. Axelrod, but it should relax its
+readiness to do so and be more \textbf{generous}. For noisy environments this is
+inline with the results of~\cite{Bendor1991, Donninger1986, Molander1985,
+Hammerstein1984}, however, we also showed that generosity pays off even in
+standard settings, and that in fact the only setting a strategy would want to be
+too provocable is when the matches are not long. Forgiveness as defined by R.
+Axerlod was not explored here. This was mainly because the two round states were
+not recorded during the data collection. That is something we would aim to
```

Comment:
Maybe simply "could be a topic of future work that examines the impact of considering more rounds of history"